From 58af9a4453c630bc3d11e88db33f334e1887e4ab Mon Sep 17 00:00:00 2001
From: Thomas Preud'homme <robotux@celest.fr>
Date: Wed, 14 Mar 2012 15:47:42 +0100
Subject: Consider long int constant as 64 bits on x86-64

Quick and dirty hack to consider long int constant (as in 1UL or 1L) as
64 bits integer on x86-64 non Windows systems.

Origin: vendor
Bug-Debian: http://bugs.debian.org/663711
Forwarded: http://repo.or.cz/w/tinycc.git/commit/1736a71b71221df7c81b0d3da25e774f3c4005d2
Last-Update: 2012-03-15
Applied-Upstream: commit: 1736a71b71221df7c81b0d3da25e774f3c4005d2
---
 tccpp.c |    4 ++++
 1 files changed, 4 insertions(+), 0 deletions(-)

diff --git a/tccpp.c b/tccpp.c
index ff47838..d1043e0 100644
--- a/tccpp.c
+++ b/tccpp.c
@@ -2028,12 +2028,16 @@ static void parse_number(const char *p)
                 if (lcount >= 2)
                     tcc_error("three 'l's in integer constant");
                 lcount++;
+#if !defined TCC_TARGET_X86_64 || defined TCC_TARGET_PE
                 if (lcount == 2) {
+#endif
                     if (tok == TOK_CINT)
                         tok = TOK_CLLONG;
                     else if (tok == TOK_CUINT)
                         tok = TOK_CULLONG;
+#if !defined TCC_TARGET_X86_64 || defined TCC_TARGET_PE
                 }
+#endif
                 ch = *p++;
             } else if (t == 'U') {
                 if (ucount >= 1)
